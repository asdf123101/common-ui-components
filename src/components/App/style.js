import styled from 'styled-components';

const Header = styled.div`
  text-align: center;
 background-color: #da3f3d;
  height: 100px;
  padding: 20px;
  color: white;
flex: 0 1 100%;
`;

const Logo = styled.img`
  height: 100px;
`;

export { Header, Logo };
