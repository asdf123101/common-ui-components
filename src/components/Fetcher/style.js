import styled from 'styled-components';

export const RefreshButton = styled.button`
  background-color: white;
  border: 2px solid #999;
  border-radius: 5px;
  padding: 8px 16px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 90%;
  margin: 4px 10px;
  width: 100px;
  outline:none;
  &:hover {
    background-color: #70C1B3;
    color: white;
    border: 2px solid #70C1B3;
    transition: background-color 0.2s, color 0.2s;
  }
`;

export const SelectButton = RefreshButton.withComponent('select').extend`
  width: auto;
  margin-left: 0;
`;

export const PostItem = styled.li`
  margin: 10px;
  border: 1px solid #ddd;
  box-shadow: 2px 2px 5px #ddd;
  padding: 10px;
  a {
    text-decoration: none;
    color: #2A5DB0 ;
  }

  a:hover {
    color: #da3f3d;
    text-decoration: auto;
    transition: color 0.2s;
  }
`;
