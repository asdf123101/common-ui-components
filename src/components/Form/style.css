input:not([type=submit]) {
    width: 150px;
    margin: 8px 10px;
    padding: 5px;
    border: none;
    border-bottom: 1px solid #ccc;
    box-sizing: border-box;
}

input:not([type=submit]):focus {
    outline: none;
    border-bottom: 1px solid #66ccff;
    box-sizing: border-box;
}
textarea:focus {
    outline: none;
    border-color: #66ccff;
}
textarea {
    width: 200px;
    height: 100px;
    margin: 8px 0;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
    display: block;
}

label {
    display: block;
}

.warning {
    display: block;
    width: 400px;
    color: red;
    font-size: 80%;
}

input[type=submit] {
    margin: 8px 0;
}

form {
    width: 500px;
}
