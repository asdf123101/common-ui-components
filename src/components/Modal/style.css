.modal {
	display: none;
}

.modal.active {
	display: block;
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	height: 100%;
	z-index: 10;
	background-color: rgba(0,0,0,0.5);
	display: flex;
	justify-content: center;
	align-items: center;
}

.modal p {
	width: 90vw;
	background-color: #fff;
	padding: 10px;
}

@media screen and (min-width: 600px) {
	.modal p {
		width: 40vw;
	}
	
}
button.hide {
	position: fixed;
	right: 20px;
	top: 20px;
}