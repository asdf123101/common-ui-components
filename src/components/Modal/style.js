import styled from 'styled-components';

const ModalWrapper = styled.div`
	display: block;
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	height: 100%;
	z-index: 10;
	background-color: rgba(0,0,0,0.5);
	display: flex;
	justify-content: center;
	align-items: center;
`;

const ModalPara = styled.p`
	width: 90vw;
	background-color: #fff;
	padding: 10px;

  @media screen and (min-width: 600px) {
		width: 40vw;
	}
`;

const ModalButton = styled.button`
	position: fixed;
	right: 20px;
	top: 20px;
`;

export { ModalWrapper, ModalPara, ModalButton };
