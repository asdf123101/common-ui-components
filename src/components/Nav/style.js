import styled from 'styled-components';

const NavUl = styled.ul`
    list-style-type: none;
    margin: 0;
    padding: 0;
    overflow: hidden;
    background-color: #333;
    height: auto;

@media  (max-width: 800px) {
        height: auto;
        width: 100vw;
}
`;

const NavLi = styled.li`
width: 100px;
a {
    display: block;
    color: white;
    text-align: center;
    padding: 14px 16px;
    text-decoration: none;
}

 a:hover {
    background-color: #111;
}

@media  (max-width: 800px) {
        float: left;
}
`;
export { NavUl, NavLi };
