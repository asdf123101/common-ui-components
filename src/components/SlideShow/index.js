import React from 'react';
import SlideShow from './SlideShow';
import SharedTitle from '../Utils/SharedTitle.jsx';

import './style.css';

const ModalDemo = () =>
  <div>
    <SharedTitle />
    <p>
      Clicking the button will slide the blocks and the center block is always
      colored. The animation is borrowed from React-flip-move library with some
      hack. Overall doing transition animation in React is a bit hard.{' '}
    </p>
    <hr />
    <SlideShow />
  </div>;

export default ModalDemo;
