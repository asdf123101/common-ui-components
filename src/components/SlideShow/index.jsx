import React from 'react'
import PropTypes from 'prop-types';
import FlipMove from 'react-flip-move';

import './style.css'

class Slide extends React.Component {
static propTypes = {
	index: PropTypes.number.isRequired,
	center: PropTypes.string.isRequired,
	zIndex: PropTypes.number,
	}
render() {
	const className = 'slide slide'+ this.props.index + this.props.center
		return (
			<div className={className} style={{zIndex: this.props.zIndex}}><h2>{this.props.slide}</h2></div>
			)
}
}


export default class SlideShow extends React.Component {
	state = {
		indices: [0,1,2,3,4],
		direction: null,
	}
	

	renderSlides = () => {
		let nSlide = this.state.indices.length
		return (
			<div className='slide-group'>
			<FlipMove duration={600} easing="ease-in-out">
			{this.state.indices.map((slide, index) => {
				const center = (index===(nSlide-1)/2) ? ' center' : ''
				let zIndex = null
				if (this.state.direction === 'left' && index === this.state.indices.length-1) {zIndex = -10}
				if (this.state.direction === 'right' && index === 0) {zIndex = -10}
				return (<Slide index={index} center={center} slide={slide} key={slide} zIndex={zIndex}/>)
			})}
			</FlipMove>
			</div>
			
		)
	}
	handleLeftClick = () => {
		let shiftedState = this.state.indices.slice(1)
		let newState = [...shiftedState,this.state.indices[0]]
		this.setState({indices: newState})
		this.setState({direction: 'left'})
	}
	
	handleRightClick = () => {
		let shiftedState = this.state.indices.slice(0,-1)
		let newState = [this.state.indices[this.state.indices.length-1],...shiftedState]
		this.setState({indices: newState})
		this.setState({direction: 'right'})
	}

	render() {
		return (
		
		<div  className='slideshow'>
			<p>Clicking the button will slide the blocks and the center block is always colored. The animation is borrowed from React-flip-move library with some hack. Overall doing transition animation in React is a bit hard. </p>
			<button className="left" onClick={this.handleLeftClick}>&lt;</button>
			{this.renderSlides()}
			<button className="right" onClick={this.handleRightClick}>&gt;</button>
		</div>
		
		)
	}
}