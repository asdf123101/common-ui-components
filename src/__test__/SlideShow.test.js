import React from 'react';
import SlideShow from '../components/SlideShow/';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Button, SingleSlide } from '../components/SlideShow/style';

describe('<SlideShow />', () => {
  it('The SlideShow has 5 slides', () => {
    const SlideShowWindow = mount(<SlideShow />);
    expect(SlideShowWindow.find(SingleSlide).length).toBe(5);
  });

  it('Snapshot testing clicking button behavior', () => {
    let SlideShowWindow = mount(<SlideShow />);
    let tree = toJson(SlideShowWindow);
    expect(tree).toMatchSnapshot();
    SlideShowWindow.find(Button).at(0).simulate('click');
    tree = toJson(SlideShowWindow);
    expect(tree).toMatchSnapshot();
    SlideShowWindow.find(Button).at(1).simulate('click');
    tree = toJson(SlideShowWindow);
    expect(tree).toMatchSnapshot();
  });

  it('Clicking on buttons add z-index style to the corresponding box', () => {
    const SlideShowWindow = mount(<SlideShow />);
    SlideShowWindow.find(Button).at(0).simulate('click');
    let Slide0 = SlideShowWindow.find(SingleSlide).at(0);
    let Slide4 = SlideShowWindow.find(SingleSlide).at(4);
    expect(Slide4.prop('zIndex')).toBeTruthy();
    expect(Slide0.prop('zIndex')).toBeFalsy();
    SlideShowWindow.find(Button).at(1).simulate('click');
    Slide0 = SlideShowWindow.find(SingleSlide).at(0);
    Slide4 = SlideShowWindow.find(SingleSlide).at(4);
    expect(Slide0.prop('zIndex')).toBeTruthy();
    expect(Slide4.prop('zIndex')).toBeFalsy();
  });
});
