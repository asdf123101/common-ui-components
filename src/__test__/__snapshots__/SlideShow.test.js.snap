// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlideShow /> Snapshot testing clicking button behavior 1`] = `
<ModalDemo>
  <div>
    <SharedTile>
      <h2>
        Comments
      </h2>
    </SharedTile>
    <p>
      Clicking the button will slide the blocks and the center block is always colored. The animation is borrowed from React-flip-move library with some hack. Overall doing transition animation in React is a bit hard.
       
    </p>
    <hr />
    <SlideShow>
      <div
        className="slideshow"
      >
        <styled.button
          onClick={[Function]}
        >
          <button
            className="sc-bdVaJa hfyYsj"
            onClick={[Function]}
          >
            &lt;
          </button>
        </styled.button>
        <div
          className="slide-group"
        >
          <FlipMovePropConverter
            delay={0}
            disableAllAnimations={false}
            duration={600}
            easing="ease-in-out"
            enterAnimation="elevator"
            getPosition={[Function]}
            leaveAnimation="elevator"
            maintainContainerHeight={false}
            staggerDelayBy={0}
            staggerDurationBy={0}
            typeName="div"
            verticalAlignment="top"
          >
            <FlipMove
              delay={0}
              delegated={
                Object {
                  "style": Object {
                    "position": "relative",
                  },
                }
              }
              disableAllAnimations={false}
              duration={600}
              easing="ease-in-out"
              enterAnimation={
                Object {
                  "from": Object {
                    "opacity": "0",
                    "transform": "scale(0)",
                  },
                  "to": Object {
                    "opacity": "",
                    "transform": "",
                  },
                }
              }
              getPosition={[Function]}
              leaveAnimation={
                Object {
                  "from": Object {
                    "opacity": "1",
                    "transform": "scale(1)",
                  },
                  "to": Object {
                    "opacity": "0",
                    "transform": "scale(0)",
                  },
                }
              }
              maintainContainerHeight={false}
              staggerDelayBy={0}
              staggerDurationBy={0}
              typeName="div"
              verticalAlignment="top"
            >
              <div
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <Slide
                  center=""
                  index={0}
                  slide={0}
                  zIndex={null}
                >
                  <styled.div
                    index={0}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH doglcs"
                    >
                      <h2>
                        0
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=""
                  index={1}
                  slide={1}
                  zIndex={null}
                >
                  <styled.div
                    index={1}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH ibyLJJ"
                    >
                      <h2>
                        1
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=" center"
                  index={2}
                  slide={2}
                  zIndex={null}
                >
                  <styled.div
                    index={2}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH fdFopt"
                    >
                      <h2>
                        2
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=""
                  index={3}
                  slide={3}
                  zIndex={null}
                >
                  <styled.div
                    index={3}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH ibyLJJ"
                    >
                      <h2>
                        3
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=""
                  index={4}
                  slide={4}
                  zIndex={null}
                >
                  <styled.div
                    index={4}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH doglcs"
                    >
                      <h2>
                        4
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
              </div>
            </FlipMove>
          </FlipMovePropConverter>
        </div>
        <styled.button
          onClick={[Function]}
        >
          <button
            className="sc-bdVaJa hfyYsj"
            onClick={[Function]}
          >
            &gt;
          </button>
        </styled.button>
      </div>
    </SlideShow>
  </div>
</ModalDemo>
`;

exports[`<SlideShow /> Snapshot testing clicking button behavior 2`] = `
<ModalDemo>
  <div>
    <SharedTile>
      <h2>
        Comments
      </h2>
    </SharedTile>
    <p>
      Clicking the button will slide the blocks and the center block is always colored. The animation is borrowed from React-flip-move library with some hack. Overall doing transition animation in React is a bit hard.
       
    </p>
    <hr />
    <SlideShow>
      <div
        className="slideshow"
      >
        <styled.button
          onClick={[Function]}
        >
          <button
            className="sc-bdVaJa hfyYsj"
            onClick={[Function]}
          >
            &lt;
          </button>
        </styled.button>
        <div
          className="slide-group"
        >
          <FlipMovePropConverter
            delay={0}
            disableAllAnimations={false}
            duration={600}
            easing="ease-in-out"
            enterAnimation="elevator"
            getPosition={[Function]}
            leaveAnimation="elevator"
            maintainContainerHeight={false}
            staggerDelayBy={0}
            staggerDurationBy={0}
            typeName="div"
            verticalAlignment="top"
          >
            <FlipMove
              delay={0}
              delegated={
                Object {
                  "style": Object {
                    "position": "relative",
                  },
                }
              }
              disableAllAnimations={false}
              duration={600}
              easing="ease-in-out"
              enterAnimation={
                Object {
                  "from": Object {
                    "opacity": "0",
                    "transform": "scale(0)",
                  },
                  "to": Object {
                    "opacity": "",
                    "transform": "",
                  },
                }
              }
              getPosition={[Function]}
              leaveAnimation={
                Object {
                  "from": Object {
                    "opacity": "1",
                    "transform": "scale(1)",
                  },
                  "to": Object {
                    "opacity": "0",
                    "transform": "scale(0)",
                  },
                }
              }
              maintainContainerHeight={false}
              staggerDelayBy={0}
              staggerDurationBy={0}
              typeName="div"
              verticalAlignment="top"
            >
              <div
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <Slide
                  center=""
                  index={0}
                  slide={1}
                  zIndex={null}
                >
                  <styled.div
                    index={0}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH doglcs"
                    >
                      <h2>
                        1
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=""
                  index={1}
                  slide={2}
                  zIndex={null}
                >
                  <styled.div
                    index={1}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH ibyLJJ"
                    >
                      <h2>
                        2
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=" center"
                  index={2}
                  slide={3}
                  zIndex={null}
                >
                  <styled.div
                    index={2}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH fdFopt"
                    >
                      <h2>
                        3
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=""
                  index={3}
                  slide={4}
                  zIndex={null}
                >
                  <styled.div
                    index={3}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH ibyLJJ"
                    >
                      <h2>
                        4
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=""
                  index={4}
                  slide={0}
                  zIndex={-10}
                >
                  <styled.div
                    index={4}
                    zIndex={-10}
                  >
                    <div
                      className="sc-bwzfXH cRmtGE"
                    >
                      <h2>
                        0
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
              </div>
            </FlipMove>
          </FlipMovePropConverter>
        </div>
        <styled.button
          onClick={[Function]}
        >
          <button
            className="sc-bdVaJa hfyYsj"
            onClick={[Function]}
          >
            &gt;
          </button>
        </styled.button>
      </div>
    </SlideShow>
  </div>
</ModalDemo>
`;

exports[`<SlideShow /> Snapshot testing clicking button behavior 3`] = `
<ModalDemo>
  <div>
    <SharedTile>
      <h2>
        Comments
      </h2>
    </SharedTile>
    <p>
      Clicking the button will slide the blocks and the center block is always colored. The animation is borrowed from React-flip-move library with some hack. Overall doing transition animation in React is a bit hard.
       
    </p>
    <hr />
    <SlideShow>
      <div
        className="slideshow"
      >
        <styled.button
          onClick={[Function]}
        >
          <button
            className="sc-bdVaJa hfyYsj"
            onClick={[Function]}
          >
            &lt;
          </button>
        </styled.button>
        <div
          className="slide-group"
        >
          <FlipMovePropConverter
            delay={0}
            disableAllAnimations={false}
            duration={600}
            easing="ease-in-out"
            enterAnimation="elevator"
            getPosition={[Function]}
            leaveAnimation="elevator"
            maintainContainerHeight={false}
            staggerDelayBy={0}
            staggerDurationBy={0}
            typeName="div"
            verticalAlignment="top"
          >
            <FlipMove
              delay={0}
              delegated={
                Object {
                  "style": Object {
                    "position": "relative",
                  },
                }
              }
              disableAllAnimations={false}
              duration={600}
              easing="ease-in-out"
              enterAnimation={
                Object {
                  "from": Object {
                    "opacity": "0",
                    "transform": "scale(0)",
                  },
                  "to": Object {
                    "opacity": "",
                    "transform": "",
                  },
                }
              }
              getPosition={[Function]}
              leaveAnimation={
                Object {
                  "from": Object {
                    "opacity": "1",
                    "transform": "scale(1)",
                  },
                  "to": Object {
                    "opacity": "0",
                    "transform": "scale(0)",
                  },
                }
              }
              maintainContainerHeight={false}
              staggerDelayBy={0}
              staggerDurationBy={0}
              typeName="div"
              verticalAlignment="top"
            >
              <div
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <Slide
                  center=""
                  index={0}
                  slide={0}
                  zIndex={-10}
                >
                  <styled.div
                    index={0}
                    zIndex={-10}
                  >
                    <div
                      className="sc-bwzfXH cRmtGE"
                    >
                      <h2>
                        0
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=""
                  index={1}
                  slide={1}
                  zIndex={null}
                >
                  <styled.div
                    index={1}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH ibyLJJ"
                    >
                      <h2>
                        1
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=" center"
                  index={2}
                  slide={2}
                  zIndex={null}
                >
                  <styled.div
                    index={2}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH fdFopt"
                    >
                      <h2>
                        2
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=""
                  index={3}
                  slide={3}
                  zIndex={null}
                >
                  <styled.div
                    index={3}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH ibyLJJ"
                    >
                      <h2>
                        3
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
                <Slide
                  center=""
                  index={4}
                  slide={4}
                  zIndex={null}
                >
                  <styled.div
                    index={4}
                    zIndex={null}
                  >
                    <div
                      className="sc-bwzfXH doglcs"
                    >
                      <h2>
                        4
                      </h2>
                    </div>
                  </styled.div>
                </Slide>
              </div>
            </FlipMove>
          </FlipMovePropConverter>
        </div>
        <styled.button
          onClick={[Function]}
        >
          <button
            className="sc-bdVaJa hfyYsj"
            onClick={[Function]}
          >
            &gt;
          </button>
        </styled.button>
      </div>
    </SlideShow>
  </div>
</ModalDemo>
`;
